src/dune_pkg/rev_store.ml

```diff
@@ -37,10 +37,15 @@ let run_capture_zero_separated_lines { dir } =
   Process.run_capture_zero_separated ~dir ~display failure_mode git
 ;;
 
-let show { dir } (Rev rev) path =
+let show { dir } revs_and_paths =
   let git = Lazy.force Vcs.git in
   let failure_mode = Vcs.git_accept () in
-  let command = [ "show"; sprintf "%s:%s" rev (Path.Local.to_string path) ] in
+  let command =
+    "show"
+    :: List.map revs_and_paths ~f:(function
+      | `Object o -> o
+      | `Path (Rev r, path) -> sprintf "%s:%s" r (Path.Local.to_string path))
```

Comment:
I worry that this probably needs to run `git show` in batches since the command line length on some platforms is [8191 characters](https://learn.microsoft.com/en-us/troubleshoot/windows-client/shell-experience/command-line-string-limitation), on others it is a more generous 2097152 characters.

Reply:
Ah yeah, that's annoying. I've added split batches to mitigate this.